tests/test-integration/src/test/java/org/eclipse/jetty/test/FailedSelectorTest.java

```diff
@@ -0,0 +1,390 @@
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.util.thread.QueuedThreadPool;
+import org.eclipse.jetty.util.thread.Scheduler;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class FailedSelectorTest
+{
+    private static final Logger LOG = Log.getLogger(FailedSelectorTest.class);
+    private HttpClient client;
+    private Server server;
+
+    @AfterEach
+    public void stopServerAndClient() throws Exception
+    {
+        server.stop();
+        client.stop();
+    }
+
+    @BeforeEach
+    public void startClient() throws Exception
+    {
+        HttpClientTransport transport = new HttpClientTransportOverHTTP(1);
+        QueuedThreadPool qtp = new QueuedThreadPool();
+        qtp.setName("Client");
+        qtp.setStopTimeout(1000);
+        client = new HttpClient(transport, null);
+        client.setExecutor(qtp);
+
+        client.setIdleTimeout(1000);
+        client.setMaxConnectionsPerDestination(1);
+        client.setMaxRequestsQueuedPerDestination(1);
+        client.start();
+    }
+
+    public void startServer(Function<Server, ServerConnector> customizeServerConsumer) throws Exception
+    {
+        server = new Server();
+        server.setStopTimeout(1000);
+        server.setStopAtShutdown(true);
+
+        ServerConnector connector = customizeServerConsumer.apply(server);
+        server.addConnector(connector);
+
+        ServletContextHandler context = new ServletContextHandler();
+        context.setContextPath("/");
+        context.addServlet(HelloServlet.class, "/hello");
+
+        ServletHolder closeHolder = new ServletHolder(new CloseSelectorServlet(connector));
+        context.addServlet(closeHolder, "/selector/close");
+
+        HandlerList handlers = new HandlerList();
+        handlers.addHandler(context);
+        handlers.addHandler(new DefaultHandler());
+
+        server.setHandler(handlers);
+
+        server.start();
+    }
+
+    @Test
+    public void testRestartServerOnSelectFailure() throws Exception
+    {
+        CountDownLatch failedLatch = new CountDownLatch(1);
+
+        startServer((server) ->
+        {
+            RestartSelectorCustomConnector connector = new RestartSelectorCustomConnector(server, 1, 1, new RestartServerTask(server, failedLatch));
+            connector.setPort(0);
+            connector.setIdleTimeout(1000);
+            return connector;
+        });
+
+        // Request /hello
+        assertRequestHello();
+
+        // Request /selector/close
+        assertRequestSelectorClose("/selector/close");
```

Comment:
A minor nit, this method has in its name the path, so there is no need to pass the path (it's always the same path).

Reply:
Done.